assets/src/edit-story/app/prepublish/guidance/media.js

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { PAGE_HEIGHT, PAGE_WIDTH } from '../../../constants';
+import getBoundRect from '../../../utils/getBoundRect';
+import { PRE_PUBLISH_MESSAGE_TYPES } from '../constants';
+
+const SAFE_ZONE_AREA = PAGE_HEIGHT * PAGE_WIDTH;
+
+const MAX_VIDEO_LENGTH_SECONDS = 60;
+const MAX_VIDEO_WIDTH = 3840;
+const MAX_VIDEO_HEIGHT = 2160;
+const MIN_VIDEO_HEIGHT = 480;
+const MIN_VIDEO_WIDTH = 852;
+
+// todo: get video frames per second
+// MIN_VIDEO_FPS = 24;
+// export function videoElementFps(element) {}
+
+/**
+ * @typedef {import('../../../types').Page} Page
+ * @typedef {import('../../../types').Element} Element
+ * @typedef {import('../types').Guidance} Guidance
+ */
+
+/**
+ * Compare an element's size to the safe zone area it is overlapping
+ * If the element takes up <50% of the safe zone, return guidance. Otherwise return undefined.
+ *
+ * @param {Element} element The element being checked for guidelines
+ * @return {Guidance|undefined} The guidance object for consumption
+ */
+export function mediaElementSizeOnPage(element) {
+  // use the bounding rectangle for rotated elements
+  const { startX, startY, endX, endY } = getBoundRect([element]);
+  // get the intersecting area of the element's rectangle and the safe zone's rectangle
+  const safeZone = {
+      left: 0,
+      right: PAGE_WIDTH,
+      bottom: PAGE_HEIGHT,
+      top: 0,
+    },
+    elemRect = {
+      left: startX,
+      right: endX,
+      bottom: endY,
+      top: startY,
+    };
+  const xOverlap = Math.max(
+    0,
+    Math.min(safeZone.right, elemRect.right) -
+      Math.max(safeZone.left, elemRect.left)
+  );
+  const yOverlap = Math.max(
+    0,
+    Math.min(safeZone.bottom, elemRect.bottom) -
+      Math.max(safeZone.top, elemRect.top)
+  );
+
+  const elementArea = xOverlap * yOverlap;
+
+  const isTooSmallOnPage = elementArea < SAFE_ZONE_AREA / 2;
+
+  if (isTooSmallOnPage) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message:
+        element.type === 'video'
+          ? __(`Video is too small on the page`, 'web-stories')
+          : __(`Image is too small on the page`, 'web-stories'),
```

Comment:
Should there be a check to ensure this function only runs on `video` and `image` elements? I added checks in all my functions to short circuit in these cases. I'm not sure if it's necessary in the end if we do all those checks in whatever controller uses these functions, though. Thoughts?

Reply:
I had the same thoughts/questions… I think we should do the check in the functions and throw an error if we get the wrong type. That way the controller can be focused on sorting/filtering elements/pages/stories and running the functions.

Reply:
Per our discussion: But let's stick with the smallest code changes for now, so we don't have to undo anything given we don't know how these functions will be called yet. Pull out type checking with the expectation that we call it on elements of the correct type.